crates/rolldown_ecmascript/src/ecma_compiler.rs

```diff
@@ -129,17 +129,17 @@ impl EcmaCompiler {
     allocator: &'a Allocator,
     program: &mut Program<'a>,
   ) -> MinifierReturn {
-    let compress_options = options.compress.unwrap_or_default();
-
-    let semantic = SemanticBuilder::new().build(program).semantic;
-    let stats = semantic.stats();
-
-    let scoping = semantic.into_scoping();
-    if run_compress {
+    let stats = if run_compress {
+      let compress_options = options.compress.unwrap_or_default();
+      let semantic = SemanticBuilder::new().build(program).semantic;
+      let stats = semantic.stats();
+      let scoping = semantic.into_scoping();
       Compressor::new(allocator).build_with_scoping(program, scoping, compress_options);
+      stats
     } else {
-      Compressor::new(allocator).dead_code_elimination(program, CompressOptions::safest());
-    }
+      Compressor::new(allocator).dead_code_elimination(program, CompressOptions::dce());
```

Comment:
Changed `safest` to `dce`. `dce` should be the correct value here because we do not want to alter syntax.